src/benchmark.cc

```diff
@@ -212,6 +213,33 @@ void State::FinishKeepRunning() {
   manager_->StartStopBarrier();
 }
 
+ThreadState::ThreadState(State& s) :
+  State(s),
+  parent_(&s)
+{
+  CHECK(!s.started_) << "Don't create a ThreadState object after measurement has started";
+  timer_ = new internal::ThreadTimer(*timer_);
```

Comment:
why is this allocated on the heap? If you're copying anyway, it could be a non-pointer member which helps avoid the explicit `delete` call in the destructor.

Reply:
This would require the ThreadTimer definition being a part of benchmark.h.

Reply:
rats.

fair point. if only we could get the API up to C++11 and use unique_ptr...